File: 128. Longest Consecutive Sequence/128. Longest Consecutive Sequence.java
import java.util.HashSet;
import java.util.Set;

class Solution {

  public int longestConsecutive(int[] nums) {
    Set<Integer> unique = new HashSet<>();
    for (int n : nums) {
      unique.add(n);
    }

    int largest = 0;

    for (int n : unique) {
      if (!unique.contains(n - 1)) {
        int current = 0;
        while (unique.contains(n)) {
          current++;
          n++;
        }
        largest = Math.max(largest, current);
      }
    }

    return largest;
  }
}

File: 128. Longest Consecutive Sequence/128. Longest Consecutive Sequence.py
from typing import List


class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        unique = set(nums)
        largest = 0

        for n in unique:
            if n - 1 not in unique:
                current = 0
                while n in unique:
                    current += 1
                    n += 1
                largest = max(largest, current)

        return largest
